tests/test_processors.py

```diff
@@ -456,6 +456,17 @@ def test_obtains_exc_info_on_True(self):
         else:
             pytest.fail("Exception not raised.")
 
+    def test_obtains_exc_info_on_number(self):
```

Comment:
I think it’s better to parametrize `test_obtains_exc_info_on_True` (and give it a better name) than to replicate almost everything.